src/drivers/plcsim_advanced/plcsim_advanced.py

```diff
@@ -0,0 +1,218 @@
+        sys.path.append(p)
+        if sys.maxsize > 2**32: # 64-Bit OS
+            clr.FindAssembly("Siemens.Simatic.Simulation.Runtime.Api.x64")
+            clr.AddReference("Siemens.Simatic.Simulation.Runtime.Api.x64")
+        else: # 32-Bit OS
+            clr.FindAssembly("Siemens.Simatic.Simulation.Runtime.Api.x86")
+            clr.AddReference("Siemens.Simatic.Simulation.Runtime.Api.x86")
+        from Siemens.Simatic.Simulation.Runtime import SimulationRuntimeManager
+except:
+    pass
+
+class plcsim_advanced(driver):
+    '''
+    Driver that can be used together with a local PLCSim Advanced Instance, using the Simulation Runtime API.
+    Parameters:
+    instanceName : The name of the PLC Sim Advanced Instance
+    '''
+
+    def __init__(self, name: str, pipe: Optional[Pipe] = None, params:dict = None):
+        """
+        :param name: (optional) Name for the driver
+        :param pipe: (optional) Pipe used to communicate with the driver thread. See gateway.py
+        """
+        # Inherit
+        driver.__init__(self, name, pipe, params)
+
+        # Parameters
+        self.instanceName = "s7-1500"
+
+
+    def connect(self) -> bool:
+        """ Connect driver.
+        
+        : returns: True if connection established False if not
+        """
+        # Make sure to send a debug message if method returns False
+        # self.sendDebugInfo('Error message here') 
+        try:
+            if SimulationRuntimeManager.RegisteredInstanceInfo.Length > 0:
+                for instance in SimulationRuntimeManager.RegisteredInstanceInfo:
+                    if instance.Name == self.instanceName:
+                        self.connection = SimulationRuntimeManager.CreateInterface(self.instanceName)
+                        if self.connection.OperatingState == EOperatingState.Run:
+                            return True
+                        self.sendDebugInfo("PLC Sim Advanced operating state is not in 'Run'")
+                self.sendDebugInfo(f"No PLC Sim Advanced instance with name {self.instanceName} found")
+            else: 
+                self.sendDebugInfo("No PLC Sim Advanced instance running")
+        except Exception as e:
+            self.sendDebugInfo(f"Connection failed, {e}")
+            
+        return False
+
+    def disconnect(self):
+        """ Disconnect driver.
+        """
+        pass
+
+
+    def addVariables(self, variables: dict):
+        """ Add variables to the driver. Correctly added variables will be added to internal dictionary 'variables'.
+        Any error adding a variable should be communicated to the server using sendDebugInfo() method.
+        : param variables: Variables to add in a dict following the setup format. (See documentation) 
+        
+        """
+        self.connection.UpdateTagList()
+
+        for var_id in list(variables.keys()):
+            var_data = dict(variables[var_id])
+            for tag in self.connection.TagInfos:
```

Comment:
The problem I see with this is that if the PLC has many tags it may take quite long to find them... It happens with OPCUA with complex projects as well, because Siemens publishes all tags. Maybe you can find a specific tag inside TagInfos?